addons/l10n_tr_nilvera_einvoice/models/account_edi_xml_ubl_tr.py

```diff
@@ -149,6 +149,20 @@ def _get_invoice_monetary_total_vals(self, invoice, taxes_vals, line_extension_a
         vals['currency_dp'] = 2
         return vals
 
+    def _get_invoice_period_vals_list(self, invoice):
+        if invoice.invoice_line_ids._fields.get('deferred_start_date'):
+            # Returns the start and end date of first invoice line since it is required that all lines must have
+            # the same start and end date.
+            line_ids = invoice.invoice_line_ids.filtered(lambda line: line.display_type == 'product')
```

Comment:
```suggestion
            line_ids = invoice.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and line.deferred_start_date)
```
Else, if your first line has not been filled, you would get False instead